include/pybind11/detail/internals.h

```diff
@@ -510,6 +514,10 @@ struct local_internals {
             = static_cast<shared_loader_life_support_data *>(ptr)->loader_life_support_tls_key;
     }
 #endif //  defined(WITH_THREAD) && PYBIND11_INTERNALS_VERSION == 4
+
+#if PYBIND11_LOCAL_INTERNALS_VERSION >= 1
+    const char *function_capsule_name = strdup("pybind11_function_capsule");
```

Comment:
Why does it need to be strdup? Doesn't making it a pointer instead of a  const char[] array break the pointer comparison again?

Reply:
Idea was to allocate memory to ensure a unique address. But this code is now gone since you are right that this should be simpler.